scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -223,3 +274,304 @@ def difc_plot2d(calib_new, calib_old=None, instr_ws=None, mask=None, vrange=(0,1
+    return mtd[outputname]
+
+# flake8: noqa: C901
+def plot_peakd(wksp: Union[str, Workspace2D], peak_positions, plot_regions=True, show_bad_cnt=True, drange=(0, 0),
+               threshold=0.01):
+    """
+    Plots peak d spacing value for each peak position in peaks
+    :param wksp: Workspace returned from collect_peaks
+    :param peak_positions: List of peak positions, or single peak position
+    :param plot_regions: Whether to show vertical lines indicating detector regions
+    :param show_bad_cnt: Whether to display number of pixels that fall outside of threshold from mean
+    :param drange: A tuple of the minimum and maximum detector ID to plot, plots full range if (0,0) or out of bounds
+    :param threshold: The fraction of the mean to display horizontal bars at +/- the mean
+    :return: plot, plot axes
+    """
+
+    ylabel = "rel strain"
+
+    peaks = peak_positions
+    if isinstance(peak_positions, float):
+        peaks = [peak_positions]
+        ylabel += " (peak {})".format(peaks[0])
+
+    if len(peaks) == 0:
+        raise ValueError("Expected one or more peak positions")
+
+    if not mtd.doesExist(str(wksp)):
+        raise ValueError("Could not find provided workspace in ADS")
+
+    fig, ax = plt.subplots()
+    ax.set_xlabel("detector IDs")
+    ax.set_ylabel(ylabel)
+
+    # Hold the mean and stddev for each peak to compute total at end
+    means = []
+    stddevs = []
+    lens = []
+
+    ax.set_prop_cycle(color=colormap_as_plot_color(len(peaks), cmap=plt.get_cmap("jet")))
+
+    regions = []
+    plotted_peaks = []
+
+    # Use full detector range if not specified
+    max_detid = int(np.max(mtd[str(wksp)].detectorInfo().detectorIDs()))
+    if drange == (0, 0) or drange > (0, max_detid):
+        drange = (0, max_detid)
+        print("Specified drange out of bounds, using {}".format(drange))
+
+    # Plot data for each peak position
+    for peak in peaks:
+        print("Processing peak position {}".format(peak))
+        single = extract_peak_info(wksp, 'single', peak)
+
+        # get x and y arrays from single peak ws
+        x = single.dataX(0)
+        y = single.dataY(0)
+
+        # filter out any nans
+        y_val = y[~np.isnan(y)]
+
+        try:
+            dstart = single.yIndexOfX(drange[0])
+        except ValueError:
+            # If detector id not valid, find next closest range
+            dstart = single.yIndexOfX(int(x.searchsorted(drange[0])) - 1)
+            print("Specified starting detector range was not valid, adjusted to detID={}".format(dstart))
+        try:
+            dend = single.yIndexOfX(drange[1])
+        except ValueError:
+            dend = single.yIndexOfX(int(x.searchsorted(drange[1])))
+            print("Specified ending detector range was not valid, adjusted to detID={}".format(dend))
+
+        if dstart <= dend:
+            cut_id = (dstart, dend)
+        else:
+            # Switch range if the start point is larger
```

Comment:
Why not throw an exception? What are the scenarios then the start point is larger than the end point, beside the case when the user made a mistake and entered a incorrect `drange`?